src/openscm_runner/adapters/fair2/_run_fair.py

```diff
@@ -0,0 +1,209 @@
+"""
+Module for running FaIR
+"""
+import logging
+
+import numpy as np
+from fair.constants.general import EARTH_RADIUS, SECONDS_PER_YEAR
+import fair.emissions_driven as fair_scm
+from scmdata import ScmRun, run_append
+
+LOGGER = logging.getLogger(__name__)
+toa_to_joule = 4 * np.pi * EARTH_RADIUS ** 2 * SECONDS_PER_YEAR
+
+
+def run_fair(cfgs, output_vars):  # pylint: disable=R0914
+    """
+    Run FaIR
+
+    Parameters
+    ----------
+    cfgs : list[dict]
+        List of configurations with which to run FaIR
+
+    output_vars : list[str]
+        Variables to output
+
+    Returns
+    -------
+    :obj:`ScmRun`
+        :obj:`ScmRun` instance with all results.
+    """
+    res = []
+
+    for cfg in cfgs:
+        scenario = cfg.pop("scenario")
+        model = cfg.pop("model")
+        run_id = cfg.pop("run_id")
+        factors = {}
+        factors["gmst"] = cfg.pop("gmst_factor")
+        factors["ohu"] = cfg.pop("ohu_factor")
+        data, unit, nt = _process_output(fair_scm(**cfg), output_vars, factors)
+
+        data_scmrun = []
+        variables = []
+        units = []
+        for key, variable in data.items():
+            variables.append(key)
+            data_scmrun.append(variable)
+            units.append(unit[key])
+
+        tempres = ScmRun(
+            np.vstack(data_scmrun).T,
+            index=np.arange(1765, 1765 + nt),
+            columns={
+                "scenario": scenario,
+                "model": model,
+                "region": "World",
+                "variable": variables,
+                "unit": units,
+                "run_id": run_id,
+            },
+        )
+
+        res.append(tempres)
+
+    res = run_append(res)
+
+    return res
+
+
+def _process_output(fair_output, output_vars, factors):
+    """
+    Make sense of FaIR2.0 output
+
+    Parameters
+    ----------
+    fair_output :  :obj:`pyam.IamDataFrame`
+        Output data
+
+    output_vars : list[str]
+        List of output variables
+
+    factors : dict[(Union[float, numpy.ndarray])]
+        ohu : ratio of ocean heat uptake to total Earth energy uptake
+        gmst : ratio of GMST to GSAT
+
+    Returns
+    -------
+    data : dict
+        dict of climate model output
+    unit : dict
+        dict of units corresponding to data
+    nt : int
+        number of timesteps modelled
+    """
+
+    dropped_df = fair_output.timeseries().droplevel(['scenario','model','region',])
+    data = {}
+    unit = {}
+    for key in dropped_df.index:
+        data[key[0]] = dropped_df.loc[key].values
+        unit[key[0]] = key[1]
+
+    data["Effective Radiative Forcing|Greenhouse Gases"] = \
+        np.sum([dropped_df[key] for key in\
+            ["Effective Radiative Forcing|" + x for x in\
+                ["CO2","CH4","N2O","CF4","C2F6","C6F14","HFC23",
```

Comment:
Is it worth storing these long-lists in a common place so all models can use them..?